### PR TITLE
Add suggested stackalloc initializer tests

### PR DESCRIPTION
**Pull request by alrz**
Add tests from feature review (see https://github.com/dotnet/roslyn/issues/25160)

### CODE REVIEW

**Comment by jcouv** [comment]
@VSadov for review (more tests for stackalloc init feature). Thanks

File: src/Compilers/CSharp/Test/Semantic/Semantics/StackAllocInitializerTests.cs [hunk 1]
```diff
@@ -1965,14 +2170,19 @@ public void Method1()
     {
         var obj1 = stackalloc double[2] { 1, 1.2 };
         Span<double> obj2 = stackalloc double[2] { 1, 1.2 };
+        _ = stackalloc double[2] { 1, 1.2 };
     }
-}", TestOptions.UnsafeReleaseDll).VerifyDiagnostics();
+}", TestOptions.UnsafeReleaseDll).VerifyDiagnostics(
+                // (9,13): error CS8353: A result of a stackalloc expression of type 'Span<double>' cannot be used in this context because it may be exposed outside of the containing method
+                //         _ = stackalloc double[2] { 1, 1.2 };
+                Diagnostic(ErrorCode.ERR_EscapeStackAlloc, "stackalloc double[2] { 1, 1.2 }").WithArguments("System.Span<double>").WithLocation(9, 13)
```
[the remark this answers is not in the document]
Comment (VSadov) [reply src/Compilers/CSharp/Test/Semantic/Semantics/StackAllocInitializerTests.cs]: "Returnable" is much more common, so it makes sense to treat ` _ ` as returnable. 
If they are considered not returnable, the cases like following would be errors:

```C#
return Method(out _ );  // what is Method returns its argument ?

Method(out _ , ref returnable);  // what is Method assigns one argument to another ?
```

**Comment by VSadov** [comment]
@dotnet-bot  test windows_coreclr_release_prtest  please

**Comment by jcouv** [comment]
Merged. Thanks @alrz!

**Comment by alrz** [comment]
I don't how this happened but TestUnmanaged_Span is failing on master. Did this even reach that branch?

**Comment by alrz** [comment]
Note: it's probably due tohttps://github.com/dotnet/roslyn/pull/24183

**Comment by jcouv** [comment]
@alrz Yes, I think it is due to [merge ](https://github.com/dotnet/roslyn/commit/2c383ed373a0463d29c3f1677a13c8f23155eaec) of stackalloc initializers (from branch dev15.7.x) with #24183 in master branch. 

PR for fix: https://github.com/dotnet/roslyn/pull/25854

**Pull request merged**